lib/path.js

```diff
@@ -1236,20 +1236,20 @@ const posix = {
   join(...args) {
     if (args.length === 0)
       return '.';
-    let joined;
+
+    const path = [];
     for (let i = 0; i < args.length; ++i) {
       const arg = args[i];
       validateString(arg, 'path');
       if (arg.length > 0) {
-        if (joined === undefined)
-          joined = arg;
-        else
-          joined += `/${arg}`;
+        path.push(arg);
```

Comment:
Does this need a primordial?

Reply:
If I use `ArrayPrototypePush`, the benchmark result is as follows.

```
                                                               confidence improvement accuracy (*)   (**)  (***)
path/join-posix.js n=100000 paths='/foo|bar||baz/asdf|quux|..'                 0.20 %       ±0.63% ±0.84% ±1.09%
```

Reply:
I think the primordial is needed here, regardless of benchmark results.

CC @nodejs/primordials

Reply:
Using primordials are not required. If we see a critical performance impact, just like this, we should avoid using it.

Reply:
How are they not required? This isn't the error path.

What about an alternative:
```suggestion
        path[path.length - 1] = arg;
```

Reply:
primordials are used elsewhere in this function, shouldn't they be used here, or removed from the other spots?